### ISSUE TITLE
RegisterObjectAsync

### ISSUE THREAD

**Issue by rquackenbush** [tmds/Tmds.DBus#13]
Hi - I'm trying to get the below to work (in order to sort out an issue I'm having with registering an object for ConnMann). Have I done something foolish below, or is this supposed to work? It's failing on the proxy.ConcatAsync call.

```
using System;
using System.Threading.Tasks;
using Tmds.DBus;
using System.Threading;

namespace ConsoleApplication
{
    public class Program
    {
        private static async Task TestAsync()
        {
            using(var connection1 = new Connection(Address.System))
            using(var connection2 = new Connection(Address.System))
            {
                await connection1.ConnectAsync();
                await connection2.ConnectAsync();

                await connection1.RegisterObjectAsync(new StringOperations());

                var proxy = connection2.CreateProxy<IStringOperations>("servicename", StringOperations.Path); 

                var result = await proxy.ConcatAsync("hi", "there", CancellationToken.None);

                Console.WriteLine(result);
            }
        }

        public static void Main(string[] args)
        {
            TestAsync().Wait();
        }
    }

    class StringOperations : IStringOperations
    {
        public static readonly ObjectPath Path = new ObjectPath("/tmds/dbus/tests/stringoperations");
        private ObjectPath _path;
        public StringOperations()
        {
            _path = Path;
        }
        public StringOperations(ObjectPath path)
        {
            _path = path;
        }
        public Task<string> ConcatAsync(string s1, string s2, CancellationToken cancellationToken)
        {
            Console.WriteLine("In ConcatAsync!");

            return Task.FromResult($"{s1}{s2}");
        }

        public ObjectPath ObjectPath { get { return _path; } }
    }

    [DBusInterface("tmds.dbus.tests.StringOperations")]
    public interface IStringOperations : IDBusObject
    {
        Task<string> ConcatAsync(string s1, string s2, CancellationToken cancellationToken);
    }
}
```

Exception (async messages stripped):

```
Tmds.DBus.DisconnectedException: Connection closed by peer ---> System.IO.IOException: Connection closed by peer
```

**Comment by tmds** [comment]
Two changes are needed:
connection1 will need to register the service name.
```C#
await connection1.RegisterServiceAsync("my.servicename");
```
"servicename" is not a valid dbus servicename, "my.servicename" is, change the CreateProxy line to:
```C#
var proxy = connection2.CreateProxy<IStringOperations>("my.servicename", StringOperations.Path);
```

At this point it should work, unless your dbus configuration doesn't allow it because of the configuration.

**Comment by rquackenbush** [comment]
That was it - thanks so much!

**Comment by tmds** [comment]
@rquackenbush I just published 0.3.0 to NuGet. One of the new things it is automatic generation of the introspection xml. Since you are publishing objects, you might like that.

Also, feel free to star the project :)
